apps/prairielearn/src/ee/lib/billing/plans.test.ts

```diff
@@ -2,71 +2,286 @@ import { assert } from 'chai';
+
+        await reconcilePlanGrantsForInstitution('1', [{ plan: 'compute', grantType: 'gift' }], '1');
+        planGrants = await getPlanGrantsForContext({ institution_id: '1' });
+        assert.isOk(planGrants.find((pg) => pg.plan_name === 'compute' && pg.type === 'gift'));
+
+        await reconcilePlanGrantsForInstitution('1', [], '1');
+        planGrants = await getPlanGrantsForContext({ institution_id: '1' });
+        assert.deepEqual(planGrants, []);
+      });
     });
   });
 
-  describe('updatePlanGrantsForCourseInstance', () => {
+  describe('reconcilePlanGrantsForCourseInstance', () => {
     it('persists updates', async () => {
-      await reconcilePlanGrantsForCourseInstance(
-        '1',
-        [{ plan: 'compute', grantType: 'invoice' }],
-        '1',
-      );
-      let planGrants = await getPlanGrantsForCourseInstance('1');
-      assert.isOk(planGrants.find((pg) => pg.plan_name === 'compute' && pg.type === 'invoice'));
-
-      await reconcilePlanGrantsForCourseInstance(
-        '1',
-        [{ plan: 'compute', grantType: 'gift' }],
-        '1',
-      );
-      planGrants = await getPlanGrantsForCourseInstance('1');
-      assert.isOk(planGrants.find((pg) => pg.plan_name === 'compute' && pg.type === 'gift'));
-
-      await reconcilePlanGrantsForCourseInstance('1', [], '1');
-      planGrants = await getPlanGrantsForCourseInstance('1');
-      assert.deepEqual(planGrants, []);
+      await runInTransactionAndRollback(async () => {
+        await reconcilePlanGrantsForCourseInstance(
+          '1',
+          [{ plan: 'compute', grantType: 'invoice' }],
+          '1',
+        );
+        let planGrants = await getPlanGrantsForCourseInstance({
+          institution_id: '1',
+          course_instance_id: '1',
+        });
+        assert.isOk(planGrants.find((pg) => pg.plan_name === 'compute' && pg.type === 'invoice'));
+
+        await reconcilePlanGrantsForCourseInstance(
+          '1',
+          [{ plan: 'compute', grantType: 'gift' }],
+          '1',
+        );
+        planGrants = await getPlanGrantsForCourseInstance({
+          institution_id: '1',
+          course_instance_id: '1',
+        });
+        assert.isOk(planGrants.find((pg) => pg.plan_name === 'compute' && pg.type === 'gift'));
+
+        await reconcilePlanGrantsForCourseInstance('1', [], '1');
+        planGrants = await getPlanGrantsForCourseInstance({
+          institution_id: '1',
+          course_instance_id: '1',
+        });
+        assert.deepEqual(planGrants, []);
+      });
+    });
+
+    it('does not modify institution plan grants', async () => {
+      await runInTransactionAndRollback(async () => {
+        // Manually insert an institution plan grant.
+        await insertPlanGrant(
+          {
+            plan_name: 'everything',
+            type: 'gift',
+            institution_id: '1',
+          },
+          '1',
```

Comment:
This second parameter is the user ID for the audit log. Maybe this should be an `authn_user_id` property on the first argument? That's more analogous to what we've started doing in PT.

Reply:
I think that would be easier to read and more likely we'd remember to get it right.